### PR TITLE
Listing merge

### PR DESCRIPTION
**Pull request by mjarco**

### CODE REVIEW

File: crdstore/crdstore.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ func (cs *CredentialsStore) updateCache(accessKey, backend, key string, csd *Cre
 }
 
 func (cs *CredentialsStore) tryLock() bool {
+	// #nosec
 	return atomic.CompareAndSwapInt32((*int32)(unsafe.Pointer(&cs.lock)), 0, 1)
```
Comment (janisz) [inline crdstore/crdstore.go]: Please add comment why it's done that way

File: crdstore/crdstore_test.go [hunk 2]
```diff
@@ -86,14 +86,14 @@ func TestShouldPrepareInternalKeyBasedOnAccessAndStorageType(t *testing.T) {
...
-	require.Equal(t, existingCredentials.SecretKey, csd.SecretKey, "key must be equal")
-}
+// func TestShouldSetCredentialsFromExternalServiceEndpoint(t *testing.T) {
+// 	cs, _ := GetInstance("default")
+
+// 	csd, err := cs.Get(existingCredentials.AccessKey, existingStorage)
+// 	require.NoError(t, err)
+// 	require.Equal(t, existingCredentials.AccessKey, csd.AccessKey, "key must be equal")
+// 	require.Equal(t, existingCredentials.SecretKey, csd.SecretKey, "key must be equal")
+// }
```
Comment (janisz) [inline crdstore/crdstore_test.go]: Delete if not needed or [`t.Skip("Why?")`](https://golang.org/pkg/testing/#T.Skip)

File: crdstore/crdstore.go (hunk 1 above)
Comment (mjarco) [reply crdstore/crdstore.go]: We'll get rid of this construct in future version. I promise :)

File: glide.yaml [hunk 3]
```diff
@@ -2,30 +2,27 @@ package: github.com/allegro/akubra
 import:
 - package: github.com/ShowMax/go-fqdn
 - package: github.com/alecthomas/kingpin
-  version: ^2.2.5
```
Comment (janisz) [inline glide.yaml]: Why you are removing versions?

File: httphandler/httphandler.go [hunk 4]
```diff
@@ -67,10 +67,13 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	defer func() {
-		closeErr := resp.Body.Close()
-		if closeErr != nil {
-			log.Printf("Cannot send response body reason: %q",
-				closeErr.Error())
+		if resp.Body != nil {
+			closeErr := resp.Body.Close()
+			if closeErr != nil {
```
Comment (janisz) [inline httphandler/httphandler.go]: Prevent nested conditions with reverting it

```go
if resp.Body == nil {
  return
}
if err := resp.Body.Close(); err != nil {
   log.Printf("Cannot send response body reason: %q", err.Error())
}
```

File: httphandler/httphandler.go [hunk 5]
```diff
@@ -80,9 +83,11 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	w.WriteHeader(resp.StatusCode)
-	if _, copyErr := io.Copy(w, resp.Body); copyErr != nil {
-		log.Printf("Cannot send response body reason: %q",
-			copyErr.Error())
+	if resp.Body != nil {
```
Comment (janisz) [inline httphandler/httphandler.go]: ditto

**Pull request merged**